### ISSUE TITLE
Upstream OVS GTP patches

### ISSUE THREAD

**Issue by fvasquez** [magma/magma#14134]
### Problem

The team I am on is trying to port Magma AGW to arm64 devices. Since Magma's OVS kernel modules rely on DKMS we can only build and deploy them on an old distro like Ubuntu 20.04 because building OVS kernel modules from the OVS source tree is deprecated as of Linux 5.8. 

The README for [Magma's GTP patches](https://github.com/magma/magma/tree/master/third_party/gtp_ovs/ovs-gtp-patches/2.15) says these patches will be removed once they are upstreamed to the OVS repo but that cannot happen now that building OVS kernel modules outside of the kernel source tree is no longer supported.

### Solution

The cleanest solution is to upstream Magma's OVS GTP patches to the official Linux kernel source repo. While this is going to be a massive effort once the task is done it won't have to be revisited. New features and maintenance will happen upstream.

Magma developers won't have to regenerate new patches to upgrade to newer Linux kernels and Magma users will no longer be shackled to older Linux kernels.

### Non-goals

Moving more OVS functionality into the Linux kernel is not the goal of this proposal. OVS has already moved GTP tunneling out of user space so Magma should follow suit. Things done best in user space should remain in user space.

**Comment by fvasquez** [comment]
@pshelar, any thoughts on the feasibility of this proposal?

**Comment by lucasgonze** [comment]
This item has high value.

**Comment by fvasquez** [comment]
I stopped working on 5G-related projects a couple of months after opening this issue. DePIN was in freefall due to the collapse of FTX. I believe the team I was on at the time eventually moved away from Magma. I managed to build a Linux kernel for arm64 with the OVS GTP patches already applied but that was no easy task. I'm sorry to say it but I think the window of opportunity for this item has passed.

**Comment by lucasgonze** [comment]
This was discussed in today's TSC meeting. The issues:

1. Magma is stuck on out of date kernels because of the need for these patches. These kernels limit security, performance, and features. The question about updating the kernel version supported can't be delayed indefinitely. 
2. The code of these patches is incompatible with recent kernels because the interfaces for network modules have changed. Being able to use recent kernels would take new code. Possibly a move to eBPF would be less work.

We need to know how best to do the work, how much work that would be, what it would cost, and who would do it. The consensus was to keep the kernel upgrade on the agenda while we think about those questions.

**Comment by lucasgonze** [comment]
Closing this in favor of https://github.com/magma/magma/issues/15454
